pkg/istio/authorizationpolicy/controller.go

```diff
@@ -369,3 +385,43 @@ func (c *Controller) checkAuthzEnabledAnnotation(serviceObj *corev1.Service) boo
 	}
 	return false
 }
+
+// cleanUpStaleAP deletes the existing Authorization Policy associated to the service which is switching back from
+// Authorization Policy Enabled back to SR/SRB
+func (c *Controller) cleanUpStaleAP() error {
```

Comment:
Can we add a unit test here? If not, can it be added as an integration test?

Reply:
Will add a unit test